src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/ActionFurtherEvidenceSubmittedCallbackHandlerTest.java

```diff
@@ -0,0 +1,119 @@
+
+import java.time.LocalDateTime;
+import java.util.Collections;
+import java.util.Optional;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import uk.gov.hmcts.reform.sscs.ccd.callback.Callback;
+import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.sscs.ccd.domain.CaseDetails;
+import uk.gov.hmcts.reform.sscs.ccd.domain.DynamicList;
+import uk.gov.hmcts.reform.sscs.ccd.domain.DynamicListItem;
+import uk.gov.hmcts.reform.sscs.ccd.domain.EventType;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseDetails;
+import uk.gov.hmcts.reform.sscs.ccd.domain.State;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.actionfurtherevidence.ActionFurtherEvidenceSubmittedCallbackHandler;
+import uk.gov.hmcts.reform.sscs.ccd.service.CcdService;
+import uk.gov.hmcts.reform.sscs.idam.IdamService;
+import uk.gov.hmcts.reform.sscs.idam.IdamTokens;
+
+@RunWith(JUnitParamsRunner.class)
+public class ActionFurtherEvidenceSubmittedCallbackHandlerTest {
+    @Rule
+    public MockitoRule rule = MockitoJUnit.rule();
+    @Mock
+    private CcdService ccdService;
+    @Mock
+    private IdamService idamService;
+
+    private ActionFurtherEvidenceSubmittedCallbackHandler handler;
+
+    @Before
+    public void setUp() {
+        handler = new ActionFurtherEvidenceSubmittedCallbackHandler(ccdService, idamService);
+    }
+
+
+    @Test
+    @Parameters(method = "generateCanHandleScenarios")
+    public void givenCanHandleIsCalled_shouldReturnCorrectResult(CallbackType callbackType,
+                                                                 Callback<SscsCaseData> callback,
+                                                                 boolean expectedResult) {
+        boolean actualResult = handler.canHandle(callbackType, callback);
+
+        assertEquals(expectedResult, actualResult);
+    }
+
+    private Object[] generateCanHandleScenarios() {
+        Callback<SscsCaseData> callbacWithRightEventAndRightField =
+            buildCallback("informationReceivedForInterloc", ACTION_FURTHER_EVIDENCE);
+        Callback<SscsCaseData> callbacWithRightEventAndWrongField =
+            buildCallback("otherDocumentManual", ACTION_FURTHER_EVIDENCE);
+        Callback<SscsCaseData> callbacWithWrongEventAndRightField =
+            buildCallback("informationReceivedForInterloc", APPEAL_RECEIVED);
+
+        CaseDetails<SscsCaseData> caseDetails = new CaseDetails<>(123L, "sscs",
+            State.INTERLOCUTORY_REVIEW_STATE, SscsCaseData.builder().build(), LocalDateTime.now());
+
+        Callback<SscsCaseData> callbacWithRightEventAndNullField = new Callback<>(caseDetails, Optional.empty(),
```

Comment:
Minor thing, but need a 'k' after 'callbac'

Reply:
really? not in Spanish hehe

Reply:
done.